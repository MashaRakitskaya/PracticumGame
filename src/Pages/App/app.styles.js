import styled from '@emotion/styled';

export const AppWrapper = styled.div`
  background: ${(props) => props.color};
`;
