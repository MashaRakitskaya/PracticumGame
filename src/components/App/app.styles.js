import styled from '@emotion/styled';

export const AppWrapper = styled.div`
  height: 100vh;
  overflow: auto;
  background-color: var(--main-bg-color);
`;
export const AppWrapperContainer = styled.div`
  display: flex;
  flex-direction: row;
  min-width: 320px;
  min-height: 100vh;
  margin: 0 auto;
`;
