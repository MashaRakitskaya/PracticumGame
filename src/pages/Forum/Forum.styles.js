import styled from '@emotion/styled';

export const ForumContainer = styled.section`
  width: 100%;
  padding-top: 12px;
`;

export const ForumTopicsList = styled.div`
  display: flex;
  flex-direction: column;
  gap: 10px;
  max-width: 750px;
`;
