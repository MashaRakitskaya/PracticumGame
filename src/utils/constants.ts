export const SIGNIN_PATH = '/signin';
export const SIGNUP_PATH = '/signup';
export const PROFILE_SETTING_PATH = '/profile-setting';
export const PASSWORD_SETTING_PATH = '/password-setting';
export const FORUM_PATH = '/forum';
export const FORUM_CHAT_ID_PATH = '/forum/:forumChatId';
export const LEADERBOARD_PATH = '/leaderboard';
export const PRESENTATION_PATH = '/presentation';
export const GAME_PATH = '/game';
export const ERROR404_PATH = '/404';
export const ERROR500_PATH = '/500';
export const ROOT_DIV = '#root';
interface LeaderboardListProps {
  id: string;
  title: string;
  points: number;
  urlImg: string;
}

interface ForumTopicsListProps {
  id: string;
  title: string;
  lastMessage: string;
  numberMessages: number;
  time: string;
  urlImg: string;
}

export const leaderboardList: LeaderboardListProps[] = [
  {
    id: '1',
    title: 'Elena List',
    points: 1246434,
    urlImg:
      'https://images.unsplash.com/photo-1518020382113-a7e8fc38eac9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=417&q=80'
  },
  {
    id: '2',
    title: 'Elena List',
    points: 100460,
    urlImg:
      'https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80'
  },
  {
    id: '11',
    title: 'Elena List',
    points: 123224,
    urlImg:
      'https://images.unsplash.com/photo-1518020382113-a7e8fc38eac9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=417&q=80'
  },
  {
    id: '21',
    title: 'Elena List',
    points: 104533200,
    urlImg:
      'https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80'
  },
  {
    id: '111',
    title: 'Elena List',
    points: 1454234,
    urlImg:
      'https://images.unsplash.com/photo-1518020382113-a7e8fc38eac9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=417&q=80'
  },
  {
    id: '211',
    title: 'Elena List',
    points: 1053500,
    urlImg:
      'https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80'
  },
  {
    id: '12',
    title: 'Elena List',
    points: 1235634,
    urlImg:
      'https://images.unsplash.com/photo-1518020382113-a7e8fc38eac9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=417&q=80'
  },
  {
    id: '22',
    title: 'Elena List',
    points: 1007740,
    urlImg:
      'https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80'
  },
  {
    id: '122',
    title: 'Elena List',
    points: 12324,
    urlImg:
      'https://images.unsplash.com/photo-1518020382113-a7e8fc38eac9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=417&q=80'
  },
  {
    id: '2222',
    title: 'Elena List',
    points: 10034420,
    urlImg:
      'https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80'
  },
  {
    id: '1244',
    title: 'Elena List',
    points: 167234,
    urlImg:
      'https://images.unsplash.com/photo-1518020382113-a7e8fc38eac9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=417&q=80'
  },
  {
    id: '2555',
    title: 'Elena List',
    points: 100240,
    urlImg:
      'https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80'
  },
  {
    id: '1676',
    title: 'Elena List',
    points: 124634,
    urlImg:
      'https://images.unsplash.com/photo-1518020382113-a7e8fc38eac9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=417&q=80'
  },
  {
    id: '268',
    title: 'Elena List',
    points: 104200,
    urlImg:
      'https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80'
  },
  {
    id: '17',
    title: 'Elena List',
    points: 134234,
    urlImg:
      'https://images.unsplash.com/photo-1518020382113-a7e8fc38eac9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=417&q=80'
  },
  {
    id: '288',
    title: 'Elena List',
    points: 104600,
    urlImg:
      'https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80'
  },
  {
    id: '1655',
    title: 'Elena List',
    points: 123456,
    urlImg:
      'https://images.unsplash.com/photo-1518020382113-a7e8fc38eac9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=417&q=80'
  },
  {
    id: '2355667',
    title: 'Elena List',
    points: 10009,
    urlImg:
      'https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80'
  }
];

export const forumTopicsList: ForumTopicsListProps[] = [
  {
    id: '23515667',
    title: 'Elena List',
    lastMessage:
      'Sed ut perspics wtrwer fwrtrgtwrtw rtwrwriat iSedSed ut perspicswtrwerfwrtrgtwrtwrtwrwriatis unde omnis isteSed ut perspicswtrwerfwrtrgtwrtwrtwrwriatis unde omnis iste ut perspicswtrwerfwrtrgtwrtwrtwrwriatis unde omnis istes unde omnis iste',
    numberMessages: 10009,
    time: '12:00',
    urlImg:
      'https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80'
  },
  {
    id: '23535667',
    title: 'Elena List',
    lastMessage: 'Sed ut perspiciatis unde omnis iste twrwriat iSedSed .',
    numberMessages: 10009,
    time: '12:00',
    urlImg:
      'https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80'
  },
  {
    id: '23556667',
    title: 'Elena List',
    lastMessage: 'Sed ut perspiciatis unde omnis iste. twrwriat iSedSed ..',
    numberMessages: 10009,
    time: '12:00',
    urlImg:
      'https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80'
  },
  {
    id: '235125667',
    title: 'Elena List',
    lastMessage: 'Sed ut perspiciatis unde omnis iste.twrwriat iSedSed ..',
    numberMessages: 10009,
    time: '12:00',
    urlImg:
      'https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80'
  },
  {
    id: '289355667',
    title: 'Elena List',
    lastMessage: 'Sed ut perspiciatis unde omnis iste... twrwriat iSedSed ',
    numberMessages: 10009,
    time: '12:00',
    urlImg:
      'https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80'
  },
  {
    id: '230855667',
    title: 'Elena List',
    lastMessage: 'Sed ut perspiciatis unde omnis iste twrwriat iSedSed ...',
    numberMessages: 10009,
    time: '12:00',
    urlImg:
      'https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80'
  },
  {
    id: '235566567',
    title: 'Elena List',
    lastMessage: 'Sed ut perspiciatis unde omnis istetwrwriat iSedSed ',
    numberMessages: 10009,
    time: '12:00',
    urlImg:
      'https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80'
  },
  {
    id: '235563567',
    title: 'Elena List',
    lastMessage: 'Sed ut perspiciatis unde omnis istetwrwriat iSedSed ',
    numberMessages: 10009,
    time: '12:00',
    urlImg:
      'https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80'
  },
  {
    id: '2355000667',
    title: 'Elena List',
    lastMessage: 'Sed ut perspiciatis unde omnis istetwrwriat iSedSed ',
    numberMessages: 10009,
    time: '12:00',
    urlImg:
      'https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80'
  },
  {
    id: '2355634567',
    title: 'Elena List',
    lastMessage: 'Sed ut perspiciatis unde omnis istetwrwriat iSedSed ',
    numberMessages: 10009,
    time: '12:00',
    urlImg:
      'https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80'
  },
  {
    id: '235556790667',
    title: 'Elena List',
    lastMessage: 'Sed ut perspiciatis unde omnis iste ',
    numberMessages: 10009,
    time: '12:00',
    urlImg:
      'https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80'
  }
];

export const messagesList = [
  { user_id: '214', id: '2514', messageText: 'fsdfsgsff' },
  { user_id: '234', id: '2174', messageText: 'fsdfsgsff' },
  {
    user_id: '2234',
    id: '2414',
    messageText:
      'fsdfs sgdf Sed ut perspiciatis unde omnis iste  cdfbd dfgd dfgdgsff'
  },
  { user_id: '24554', id: '275714', messageText: 'fsdfsg dfgd dsgd drgdesff' },
  { user_id: '26574', id: '21884', messageText: 'fsd dfgdb fsgsff' },
  { user_id: '25784', id: '29914', messageText: 'fsdfvcbdgfhdbg sgsff' },
  {
    user_id: '2334',
    id: '21465',
    messageText: 'fsdfcdfgdfgb   cfbdfgbdfbd cghfbfsgsff'
  },
  { user_id: '25784', id: '2994514', messageText: 'fsdfvcbdgfhdbg sgsff' },
  { user_id: '214', id: '25rt14', messageText: 'fsdfsgsff' },
  { user_id: '234', id: '21y4674', messageText: 'fsdfsgsff' },
  {
    user_id: '2234',
    id: '24e514',
    messageText:
      'fsdfs sgdf Sed ut perspiciatis unde omnis iste  cdfbd dfgd dfgdgsff'
  },
  {
    user_id: '24554',
    id: '27546714',
    messageText: 'fsdfsg dfgd dsgd drgdesff'
  },
  { user_id: '26574', id: '2182484', messageText: 'fsd dfgdb fsgsff' },
  { user_id: '25784', id: '2991514', messageText: 'fsdfvcbdgfhdbg sgsff' },
  {
    user_id: '2334',
    id: '2146375',
    messageText: 'fsdfcdfgdfgb   cfbdfgbdfbd cghfbfsgsff'
  },
  { user_id: '25784', id: '29945714', messageText: 'fsdfvcbdgfhdbg sgsff' }
];
