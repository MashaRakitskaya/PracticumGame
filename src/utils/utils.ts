export const convertDateToLocaleString = (createdAt: string | undefined) => {
  if (createdAt) return new Date(createdAt).toLocaleString();
};
