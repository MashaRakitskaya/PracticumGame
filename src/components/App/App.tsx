import React, { useState } from 'react';
import { Route, Routes } from 'react-router-dom';

import Leaderboard from '../../pages/Leaderboard/Leaderboard';
import Sidebar from '../Sidebar/Sidebar';
import { AppWrapper, AppWrapperContainer } from './app.styles';
import {
  LEADERBOARD_PATH,
  FORUM_PATH,
  FORUM_CHAT_ID_PATH
} from '../../utils/constants';
import Forum from '../../pages/Forum/Forum';
import ForumChat from '../ForumChat/ForumChat';
import Popup from '../Popup/Popup';

function App() {
  const [isCreateTopicPopupOpen, setCreateTopicPopupOpen] = useState(false);

  const closePopup = () => {
    setCreateTopicPopupOpen(false);
  };

  const showPopup = () => {
    setCreateTopicPopupOpen(true);
  };

  function closeByOverlay(event: any): void {
    if (event.target.classList.contains('popup')) {
      closePopup();
    }
  }

  return (
    <AppWrapper>
      <AppWrapperContainer>
        <Sidebar showPopup={showPopup} />
        <Routes>
          <Route path={LEADERBOARD_PATH} element={<Leaderboard />} />
          <Route path={FORUM_PATH} element={<Forum />} />
          <Route path={FORUM_CHAT_ID_PATH} element={<ForumChat />} />
        </Routes>

        <Popup
          isOpen={isCreateTopicPopupOpen}
          closeByOverlay={closeByOverlay}
          title={'Create topic'}
          buttonText={'Create'}
          buttonType={'submit'}
        >
          <input></input>
        </Popup>
      </AppWrapperContainer>
    </AppWrapper>
  );
}

export default App;
