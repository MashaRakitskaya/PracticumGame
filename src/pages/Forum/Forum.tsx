import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import ForumChat from '../../components/ForumChat/ForumChat';
import ForumTopicItem from '../../components/ForumTopicItem/ForumTopicItem';
import { forumTopicsList, FORUM_PATH } from '../../utils/constants';
import { ForumContainer, ForumTopicsList } from './Forum.styles';

const Forum = () => {
  const [isTopicSelected, setTopicSelected] = useState(false);
  let navigate = useNavigate();
  return (
    <ForumContainer>
      <ForumTopicsList>
        {forumTopicsList.map((item) => {
          return (
            <ForumTopicItem
              onTopicClick={() => {
                setTopicSelected(true);
                navigate(`${FORUM_PATH}/${item.id}`);
              }}
              forumTopicsItem={item}
              key={item.id}
            />
          );
        })}
      </ForumTopicsList>
      {isTopicSelected && <ForumChat />}
    </ForumContainer>
  );
};

export default Forum;
