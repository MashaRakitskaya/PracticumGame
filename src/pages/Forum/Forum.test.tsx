import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Forum from './Forum';
import userEvent from '@testing-library/user-event';
import { FORUM_PATH } from '../../utils/constants';
import App from '../../components/App/App';

describe('Forum component', () => {
  it('render Forum', () => {
    render(<Forum />, { wrapper: BrowserRouter });
    expect(screen.getByRole('forum')).toBeInTheDocument();
  });

  it('get ForumChat on ForumTopicItem click', async () => {
    window.history.pushState({}, 'Forum page', FORUM_PATH);
    render(<Forum />, { wrapper: BrowserRouter });

    const topicItem = screen.getAllByRole('topic');
    userEvent.click(topicItem[0]);

    await waitFor(() => {
      expect(screen.getByRole('forumChat')).toBeInTheDocument();
    });
  });

  it('open Popup on click button add topic', async () => {
    window.history.pushState({}, 'Forum page', FORUM_PATH);
    render(<App />);

    userEvent.click(
      screen.getByRole('button', {
        name: /add topic/i
      })
    );

    await waitFor(() => {
      expect(
        screen.getByRole('heading', {
          name: /create topic/i
        })
      ).toBeInTheDocument();
    });
  });
});
