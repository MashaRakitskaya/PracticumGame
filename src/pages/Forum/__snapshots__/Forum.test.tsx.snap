// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forum component get ForumChat on ForumTopicItem click 1`] = `
<section
  class="css-jgqqyw"
  role="forumChat"
>
  <div
    class="css-z3wvmw"
  >
    <button
      class="css-1kykoac"
      type="button"
    />
    <div
      class="css-14zx4vh"
    >
      <img
        class="css-e1msmy"
        src="https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80"
      />
      <h3
        class="css-b03ueb"
      >
        Elena List
      </h3>
    </div>
    <button
      class="css-143ihu4"
    >
      delete chat
    </button>
  </div>
  <ul
    class="css-1rl014u"
  >
    <li
      class="css-1dyawmy"
    >
      fsdfsgsff
    </li>
    <li
      class="css-1dyawmy"
    >
      fsdfsgsff
    </li>
    <li
      class="css-1dyawmy"
    >
      fsdfs sgdf Sed ut perspiciatis unde omnis iste  cdfbd dfgd dfgdgsff
    </li>
    <li
      class="css-1dyawmy"
    >
      fsdfsg dfgd dsgd drgdesff
    </li>
    <li
      class="css-1dyawmy"
    >
      fsd dfgdb fsgsff
    </li>
    <li
      class="css-or1jv2"
    >
      fsdfvcbdgfhdbg sgsff
    </li>
    <li
      class="css-1dyawmy"
    >
      fsdfcdfgdfgb   cfbdfgbdfbd cghfbfsgsff
    </li>
    <li
      class="css-or1jv2"
    >
      fsdfvcbdgfhdbg sgsff
    </li>
    <li
      class="css-1dyawmy"
    >
      fsdfsgsff
    </li>
    <li
      class="css-1dyawmy"
    >
      fsdfsgsff
    </li>
    <li
      class="css-1dyawmy"
    >
      fsdfs sgdf Sed ut perspiciatis unde omnis iste  cdfbd dfgd dfgdgsff
    </li>
    <li
      class="css-1dyawmy"
    >
      fsdfsg dfgd dsgd drgdesff
    </li>
    <li
      class="css-1dyawmy"
    >
      fsd dfgdb fsgsff
    </li>
    <li
      class="css-or1jv2"
    >
      fsdfvcbdgfhdbg sgsff
    </li>
    <li
      class="css-1dyawmy"
    >
      fsdfcdfgdfgb   cfbdfgbdfbd cghfbfsgsff
    </li>
    <li
      class="css-or1jv2"
    >
      fsdfvcbdgfhdbg sgsff
    </li>
  </ul>
  <form
    class="css-2uruz5"
  >
    <input
      autocomplete="off"
      class="css-1rzlyxe"
      name="message"
      placeholder="Write a message..."
      type="text"
    />
    <button
      class="css-ycf2bp"
      type="submit"
    />
  </form>
</section>
`;

exports[`Forum component render Forum 1`] = `
<DocumentFragment>
  <section
    class="css-s867uj"
  >
    <div
      class="css-1bok71w"
    >
      <article
        class="css-1lq786t"
        role="topic"
      >
        <img
          alt="topic img"
          class="css-1ibtaqc"
          src="https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80"
        />
        <div
          class="css-1635o7l"
        >
          <h3
            class="css-f915ow"
          >
            Elena List
          </h3>
          <p
            class="css-1e7sxse"
          >
            Sed ut perspics wtrwer fwrtrgtwrtw rtwrwriat iSedSed ut perspicswtrwerfwrtrgtwrtwrtwrwriatis unde omnis isteSed ut perspicswtrwerfwrtrgtwrtwrtwrwriatis unde omnis iste ut perspicswtrwerfwrtrgtwrtwrtwrwriatis unde omnis istes unde omnis iste
          </p>
        </div>
        <div
          class="css-pntszt"
        >
          <time
            class="css-6zolz9"
          >
            12:00
          </time>
          <p
            class="css-uponyz"
          >
            10009
          </p>
        </div>
      </article>
      <article
        class="css-1lq786t"
        role="topic"
      >
        <img
          alt="topic img"
          class="css-1ibtaqc"
          src="https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80"
        />
        <div
          class="css-1635o7l"
        >
          <h3
            class="css-f915ow"
          >
            Elena List
          </h3>
          <p
            class="css-1e7sxse"
          >
            Sed ut perspiciatis unde omnis iste twrwriat iSedSed .
          </p>
        </div>
        <div
          class="css-pntszt"
        >
          <time
            class="css-6zolz9"
          >
            12:00
          </time>
          <p
            class="css-uponyz"
          >
            10009
          </p>
        </div>
      </article>
      <article
        class="css-1lq786t"
        role="topic"
      >
        <img
          alt="topic img"
          class="css-1ibtaqc"
          src="https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80"
        />
        <div
          class="css-1635o7l"
        >
          <h3
            class="css-f915ow"
          >
            Elena List
          </h3>
          <p
            class="css-1e7sxse"
          >
            Sed ut perspiciatis unde omnis iste. twrwriat iSedSed ..
          </p>
        </div>
        <div
          class="css-pntszt"
        >
          <time
            class="css-6zolz9"
          >
            12:00
          </time>
          <p
            class="css-uponyz"
          >
            10009
          </p>
        </div>
      </article>
      <article
        class="css-1lq786t"
        role="topic"
      >
        <img
          alt="topic img"
          class="css-1ibtaqc"
          src="https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80"
        />
        <div
          class="css-1635o7l"
        >
          <h3
            class="css-f915ow"
          >
            Elena List
          </h3>
          <p
            class="css-1e7sxse"
          >
            Sed ut perspiciatis unde omnis iste.twrwriat iSedSed ..
          </p>
        </div>
        <div
          class="css-pntszt"
        >
          <time
            class="css-6zolz9"
          >
            12:00
          </time>
          <p
            class="css-uponyz"
          >
            10009
          </p>
        </div>
      </article>
      <article
        class="css-1lq786t"
        role="topic"
      >
        <img
          alt="topic img"
          class="css-1ibtaqc"
          src="https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80"
        />
        <div
          class="css-1635o7l"
        >
          <h3
            class="css-f915ow"
          >
            Elena List
          </h3>
          <p
            class="css-1e7sxse"
          >
            Sed ut perspiciatis unde omnis iste... twrwriat iSedSed 
          </p>
        </div>
        <div
          class="css-pntszt"
        >
          <time
            class="css-6zolz9"
          >
            12:00
          </time>
          <p
            class="css-uponyz"
          >
            10009
          </p>
        </div>
      </article>
      <article
        class="css-1lq786t"
        role="topic"
      >
        <img
          alt="topic img"
          class="css-1ibtaqc"
          src="https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80"
        />
        <div
          class="css-1635o7l"
        >
          <h3
            class="css-f915ow"
          >
            Elena List
          </h3>
          <p
            class="css-1e7sxse"
          >
            Sed ut perspiciatis unde omnis iste twrwriat iSedSed ...
          </p>
        </div>
        <div
          class="css-pntszt"
        >
          <time
            class="css-6zolz9"
          >
            12:00
          </time>
          <p
            class="css-uponyz"
          >
            10009
          </p>
        </div>
      </article>
      <article
        class="css-1lq786t"
        role="topic"
      >
        <img
          alt="topic img"
          class="css-1ibtaqc"
          src="https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80"
        />
        <div
          class="css-1635o7l"
        >
          <h3
            class="css-f915ow"
          >
            Elena List
          </h3>
          <p
            class="css-1e7sxse"
          >
            Sed ut perspiciatis unde omnis istetwrwriat iSedSed 
          </p>
        </div>
        <div
          class="css-pntszt"
        >
          <time
            class="css-6zolz9"
          >
            12:00
          </time>
          <p
            class="css-uponyz"
          >
            10009
          </p>
        </div>
      </article>
      <article
        class="css-1lq786t"
        role="topic"
      >
        <img
          alt="topic img"
          class="css-1ibtaqc"
          src="https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80"
        />
        <div
          class="css-1635o7l"
        >
          <h3
            class="css-f915ow"
          >
            Elena List
          </h3>
          <p
            class="css-1e7sxse"
          >
            Sed ut perspiciatis unde omnis istetwrwriat iSedSed 
          </p>
        </div>
        <div
          class="css-pntszt"
        >
          <time
            class="css-6zolz9"
          >
            12:00
          </time>
          <p
            class="css-uponyz"
          >
            10009
          </p>
        </div>
      </article>
      <article
        class="css-1lq786t"
        role="topic"
      >
        <img
          alt="topic img"
          class="css-1ibtaqc"
          src="https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80"
        />
        <div
          class="css-1635o7l"
        >
          <h3
            class="css-f915ow"
          >
            Elena List
          </h3>
          <p
            class="css-1e7sxse"
          >
            Sed ut perspiciatis unde omnis istetwrwriat iSedSed 
          </p>
        </div>
        <div
          class="css-pntszt"
        >
          <time
            class="css-6zolz9"
          >
            12:00
          </time>
          <p
            class="css-uponyz"
          >
            10009
          </p>
        </div>
      </article>
      <article
        class="css-1lq786t"
        role="topic"
      >
        <img
          alt="topic img"
          class="css-1ibtaqc"
          src="https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80"
        />
        <div
          class="css-1635o7l"
        >
          <h3
            class="css-f915ow"
          >
            Elena List
          </h3>
          <p
            class="css-1e7sxse"
          >
            Sed ut perspiciatis unde omnis istetwrwriat iSedSed 
          </p>
        </div>
        <div
          class="css-pntszt"
        >
          <time
            class="css-6zolz9"
          >
            12:00
          </time>
          <p
            class="css-uponyz"
          >
            10009
          </p>
        </div>
      </article>
      <article
        class="css-1lq786t"
        role="topic"
      >
        <img
          alt="topic img"
          class="css-1ibtaqc"
          src="https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80"
        />
        <div
          class="css-1635o7l"
        >
          <h3
            class="css-f915ow"
          >
            Elena List
          </h3>
          <p
            class="css-1e7sxse"
          >
            Sed ut perspiciatis unde omnis iste 
          </p>
        </div>
        <div
          class="css-pntszt"
        >
          <time
            class="css-6zolz9"
          >
            12:00
          </time>
          <p
            class="css-uponyz"
          >
            10009
          </p>
        </div>
      </article>
    </div>
  </section>
</DocumentFragment>
`;
